src/ioctls/vcpu.rs

```diff
@@ -1162,45 +1168,61 @@ impl VcpuFd {
     /// # Arguments
     ///
     /// * `reg_id` - ID of the register for which we are setting the value.
-    /// * `data` - value for the specified register.
+    /// * `data` - byte slice where the register value will be written to.
+    ///
+    /// # Note
+    ///
+    /// `data` should be equal or bigger then the register size
+    /// oterwise function will return EINVAL error
     #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
-    pub fn set_one_reg(&self, reg_id: u64, data: u128) -> Result<()> {
-        let data_ptr = &data as *const _;
+    pub fn set_one_reg(&self, reg_id: u64, data: &[u8]) -> Result<usize> {
```

Comment:
Mh, for the second point, what ARM CPUs do we have in the rust-vmm CI? Afaik only Gravitron 3 (correct me if I'm wrong here @ShadowCurse) have any 256 bit registers.

Reply:
We are currently running on C6g.metal, but we can easily switch to C7g.metal which is Graviton 3. The test though doesn't need to fail in case the registers are not there because we should be able to run them on different hardware as well.

Reply:
@andreeaflorescu I did some testing regarding register width on c6g and c7g and found out that on both system the max register length available through KVM is 128 bits. (for some reason SVE registers are not provided)
I added error cases to `test_get/set_one_reg` tests with already present 64 bit register.

Reply:
Hmm, but then I am confused, why do we need this patch? Is this not supported by KVM?